src/Core/AspectKernel.php

```diff
@@ -254,7 +284,7 @@ protected function registerTransformers()
                 $transformers[] = new ConstructorExecutionTransformer();
             }
 
-            return $transformers;
+            return $aspectKernel->customizeTransformers($transformers);
```

Comment:
I understood your vision, but your method doesn't add significant improvements in handling of custom extensions, compared to the initial way, so I decided to reject it :)